### PR TITLE
Added bmxx80 and Iot.Device.Common.WeatherHelper

### PR DESCRIPTION
**Pull request by TechPreacher**
## Description

- Implements Bmxx80 device.
- Added Iot.Device.Common.WeatherHelper

## Motivation and Context

- Implements https://github.com/Ellerbach/nanoFrameworkIoT/projects/1#card-61501111
- Implements https://github.com/Ellerbach/nanoFrameworkIoT/projects/1#card-61565811

## How Has This Been Tested?

- Unable to test with local hardware.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Unit Tests (work on Unit Tests, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: devices/Bmxx80/Bme280.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        /// Gets or sets the humidity sampling.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <see cref="Sampling"/> is set to an undefined mode.</exception>
+        [Property]
+        public Sampling HumiditySampling
+        {
+            get => _humiditySampling;
+            set
+            {
+                if (!value.Equals(Sampling.HighResolution) &&
```
Comment (Ellerbach) [inline devices/Bmxx80/Bme280.cs:55]: I've been more brutal than that :-) I just removed the check of IsDefined!

File: devices/Bmxx80/ListBme680HeaterProfileConfig.cs [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+        /// contains elements copied from the specified collection and has sufficient capacity
+        /// to accommodate the number of elements copied.
+        /// </summary>
+        /// <param name="collection">The collection whose elements are copied to the new list.</param>
+        /// <exception cref="System.ArgumentNullException">collection is null</exception>
+        public ListBme680HeaterProfileConfig(IEnumerable collection)
+        {
+            if (collection == null)
+            {
+                throw new ArgumentNullException($"collection is null");
```
Comment (Ellerbach) [inline devices/Bmxx80/ListBme680HeaterProfileConfig.cs:41]: you can remove the text in all exception, gain of space on nano ;-)

File: devices/Bmxx80/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+3 examples on how to use this device binding are available in the [samples](samples) folder.
+
+The following fritzing diagram illustrates one way to wire up the BMP280 with a Raspberry Pi using I2C:
+
+![Raspberry Pi Breadboard diagram](samples/rpi-bmp280_i2c.png)
+
+General:
+
+| Bmp280 | Raspberry |
```
Comment (Ellerbach) [inline devices/Bmxx80/README.md:26]: you could change Raspberry for MCU ;-)

File: src/Iot.Device.Common/WeatherHelper.cs [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+        {
+            double pa = CalculateActualVaporPressure(airTemperature, relativeHumidity).Hectopascals;
+            double a = 6.1121; // hPa
+            double c = 257.14; // °C
+            double b = 18.678;
+            double dewPoint = (c * Math.Log(pa / a)) / (b - Math.Log(pa / a));
+            return Temperature.FromDegreesCelsius(dewPoint);
+        }
+
+        /*
```
Comment (Ellerbach) [inline src/Iot.Device.Common/WeatherHelper.cs:129]: Maybe just add a note why it's commented, with a TODO: once density will be present, uncomment this part

**Review by Ellerbach — APPROVED** [review]
Few comments, all up, great work!

**Comment by josesimoes** [comment]
If you add the bit for nbgv the build will pass.

Let me know if you want me to do that, as I've been fixing it for the others and know what to change. 😉

File: src/Iot.Device.Common/WeatherHelper.cs (hunk 4 above)
Comment (TechPreacher) [reply src/Iot.Device.Common/WeatherHelper.cs:129]: Great point. added.

File: devices/Bmxx80/README.md (hunk 3 above)
Comment (TechPreacher) [reply devices/Bmxx80/README.md:26]: Done.

File: devices/Bmxx80/ListBme680HeaterProfileConfig.cs (hunk 2 above)
Comment (TechPreacher) [reply devices/Bmxx80/ListBme680HeaterProfileConfig.cs:41]: Changed all.

File: devices/Bmxx80/Bme280.cs (hunk 1 above)
Comment (TechPreacher) [reply devices/Bmxx80/Bme280.cs:55]: Ok, but as I already converted it, I'll leave the checks in, I guess.

**Comment by TechPreacher** [comment]
> If you add the bit for nbgv the build will pass.
> 
> Let me know if you want me to do that, as I've been fixing it for the others and know what to change. 😉

Yes, that would be awesome. You can also show me how to do it - I'm unsure right now.

**Pull request merged**